### PR TITLE
aws secret manager secret

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_ec2_transit_gateway_attachment.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class AwsEc2TransitGatewayAttachment < AwsResourceBase
   desc 'Describes one or more attachments between resources and transit gateways. By default, all attachments are described. Alternatively, you can filter the results by attachment ID, attachment state, resource ID, or resource owner.'
 
   example "
-    describe aws_ec2_transit_gateway_attachement(transit_gateway_attachment_id: 'tgw-attach-0123456789') do
```
Comment (Nirbhay1997) [inline libraries/aws_ec2_transit_gateway_attachment.rb:10]: Please check

File: libraries/aws_secretsmanager_secrets.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    @table = fetch_data
+  end
+
+  def fetch_data
+    rows = []
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.secretsmanager_client.list_secrets(@query_params)
+      end
+      return [] if !@api_response || @api_response.empty?
```
Comment (Nirbhay1997) [inline libraries/aws_secretsmanager_secrets.rb:49]: we can return rows

**Review by Nirbhay1997 — APPROVED** [review]

**Review by Nirbhay1997 — APPROVED** [review]
LGTM

**Comment by sathish-progress** [comment]
please update readme

**Review by IanMadd — APPROVED** [review]

File: libraries/aws_secretsmanager_secrets.rb (hunk 2 above)
Comment (soumyo13) [reply libraries/aws_secretsmanager_secrets.rb:49]: Done

**Pull request merged**